operator/v1/types_dns.go

```diff
@@ -0,0 +1,40 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterDNS manages CoreDNS component to provide a name resolution service
+// in the cluster.
+type ClusterDNS struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ClusterDNSSpec   `json:"spec,omitempty"`
+	Status ClusterDNSStatus `json:"status,omitempty"`
+}
+
+// ClusterDNSSpec is the specification of the desired behavior of the ClusterDNS.
+type ClusterDNSSpec struct {
+}
+
+// ClusterDNSStatus defines the observed status of the ClusterDNS.
+type ClusterDNSStatus struct {
+	// ClusterIP is the service IP reserved for cluster DNS service
+	ClusterIP string `json:"clusterIP"`
```

Comment:
I suspect you'll prefer to have a stanza holding things like this.

Reply:
Did you have some ideas in mind about what that might be named and what else it might contain?